### DISCUSSION TITLE
How do I find out the issue with  'next build'

### DISCUSSION THREAD

**Discussion by landed1** [vercel/next.js#74437] [Help] [answered]
### Summary

I am unable to build a project locally. 
![image](https://github.com/user-attachments/assets/35818a99-7fdd-4aac-8b42-07f483168cfe)
How do I go about finding out what is wrong. The yellow warning I feel is not the issue and I saw that was related to node v23 vs 22 mostly. I am able to build a new fresh install of NextJS 15. So this makes me question the project construction.

Thank You

### Additional information

```js
I have tried removing node_modules and 'pnpm install' again - no effect.
```

### Example

https://github.com/landed1/laptop-reviews

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Not sure why your terminal is not printing all errors, but you have an issue with `app/[cat]/page.tsx`:

```console
app/[cat]/page.tsx
Type error: Type '{ params: { cat: string; }; }' does not satisfy the constraint 'PageProps'.
  Types of property 'params' are incompatible.
    Type '{ cat: string; }' is missing the following properties from type 'Promise<any>': then, catch, finally, [Symbol.toStringTag]
```

Do these:

```ts
type PostParams = {
  params: Promise<{ cat: string }>;
  searchParams?: Promise<{ [key: string]: string | string[] | undefined }>;
};

// and then further down

export async function generateMetadata({
  params,
}: {
  params: Promise<{ cat: string }>; // -> here, or better yet
}) {

// do this for generateMetadata
export async function generateMetadata({
  params,
}:  Pick<PostParams, 'params'>) {
```

See it live here, https://stackblitz.com/edit/github-ybuhz3ly?file=app%2F[cat]%2Fpage.tsx

**Reply by landed1** [reply]
Thank you yes I saw these errors when trying to deploy to GitHub. But frustrated as to why it doesn't show locally.
I will go through your answer in due course. I had this issue on reddit Next.JS group and no solution yet. I THANK YOU H New(Year)

**Reply by icyJoseph** [reply]
No worries. Maybe the street of your terminal is not working, for some reason?

Remember that your route definition can only have one cat, so to speak 

To be fair, I have helped quite a lot of people who run into this issue. Hopefully, the next.js Typescript plugin can start to flag them directly into the IDE, sooner rather than later. Have a good one!

And, please, don't forget to mark as answered if it works out well for you.

**Reply by landed1** [reply]
I went through a clean build to try and find where the build stops working and it looks like here:
pnpm next build        
(node:25904) ExperimentalWarning: CommonJS module /Users/calvincrane/dev/next15/next.config.compiled.js is loading ES Module /Users/calvincrane/dev/next15/node_modules/.pnpm/remark-gfm@4.0.0/node_modules/remark-gfm/index.js using require().
Support for loading ES Module in require() is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)

**Comment by XoL1507** [comment]
Find to connect personally. It's our rights to connect not only by net and find out to think not like mine but also by yours.
